dissect/target/plugins/os/unix/linux/fortios/_os.py

```diff
@@ -0,0 +1,382 @@
+
+        # FortiGate
+        if (datafs_tar := sysvol.path("/datafs.tar.gz")).exists():
+            target.fs.add_layer().mount("/data", TarFilesystem(datafs_tar.open("rb")))
+
+        # Additional FortiGate tars with corrupt XZ streams
+        for path in ("bin.tar.xz", "usr.tar.xz", "migadmin.tar.xz", "node-scripts.tar.xz"):
+            if (tar := target.fs.path(path)).exists():
+                fh = xz.repair_checksum(tar.open("rb"))
+                target.fs.add_layer().mount("/", TarFilesystem(fh))
+
+        # FortiAnalyzer
+        if (rootfs_ext_tar := sysvol.path("rootfs-ext.tar.xz")).exists():
+            target.fs.add_layer().mount("/", TarFilesystem(rootfs_ext_tar.open("rb")))
+
+        # Filesystem mounts can be discovered in the FortiCare debug report
+        # or using ``fnsysctl ls`` and ``fnsysctl df`` in the cli.
+        for fs in target.filesystems:
+            # log partition
+            if fs.__type__ == "ext" and (
+                fs.extfs.volume_name.startswith("LOGUSEDX") or fs.path("/root/clog").is_symlink()
+            ):
+                target.fs.mount("/var/log", fs)
+
+            # EFI partition
+            if fs.__type__ == "fat" and fs.path("/EFI").exists():
+                target.fs.mount("/boot", fs)
+
+            # data2 partition
+            if fs.__type__ == "ext" and fs.path("/new_alert_msg").exists() and fs.path("/template").exists():
+                target.fs.mount("/data2", fs)
+
+        return cls(target)
+
+    @export(property=True)
+    def hostname(self) -> str | None:
+        """Return configured hostname."""
+        try:
+            return self._config["global-config"]["system"]["global"]["hostname"][0]
+        except KeyError:
+            return None
+
+    @export(property=True)
+    def ips(self) -> list[str]:
+        """Return IP addresses of configured interfaces."""
+        result = []
+
+        try:
+            # FortiOS 6 and 7 (from global_system_interface.gz)
+            for key, iface in self._config["interfaces"].items():
+                if not key.startswith("port"):
+                    continue
+                result += [ip for ip in iface["ip"] if not ip.startswith("255")]
+        except KeyError as e:
+            self.target.log.debug("Exception while parsing FortiOS interfaces", exc_info=e)
+
+        try:
+            # Other versions
+            for conf in self._config["global-config"]["system"]["interface"].values():
+                if "ip" in conf:
+                    result.append(conf.ip[0])
+        except KeyError as e:
+            self.target.log.debug("Exception while parsing FortiOS system interfaces", exc_info=e)
+
+        return result
+
+    @export(property=True)
+    def dns(self) -> list[str]:
+        """Return configured WAN DNS servers."""
+        entries = []
+        for _, entry in self._config["global-config"]["system"]["dns"].items():
+            entries.append(entry[0])
+        return entries
```

Comment:
This is not an OSPlugin method and should be put somewhere else.

Reply:
This currently overrides from `LinuxPlugin`, so I propose to leave it be for now, until we do a big OSPlugin cleanup.

Reply:
I see, that is nasty.